File: CPE/Exam/231017/23F2_UVA10093.py
"""
    進制轉換
    base-10 下若一個數字的數位和為 9 的倍數，則該數字也是9的倍數
    可推廣到 base-r 下，若一個數字的每位總和為 r-1 的倍數，則該數字也是 r-1 的倍數

    翻了比較古老的翻譯，測資可能是包含非合法字元的，所以要先判斷是不是英文或數字
"""

while True:
    try:
        line = input().strip()
    except EOFError:
        break
    A = [] # 逐位轉換成數字
    for ch in line:
        if ch.isdigit(): # [0, 9]
            A.append(int(ch))
        elif ch.isupper(): # [10, 35]
            A.append(ord(ch) - ord("A") + 10)
        elif ch.islower(): # [36, 61]
            A.append(ord(ch) - ord("a") + 36)

    ans = -1
    r = max(A) + 1 # 最小必須為 base-r
    s = sum(A) 
    for i in range(max(2, r), 63):
        if s % (i-1) == 0:
            ans = i
            break
    print(ans if ans != -1 else "such number is impossible!")

File: CPE/Exam/231017/23F2_UVA10093.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int ans, s, mx;
    string line;
    while (cin >> line) {
        vector<int> A;
        for (char ch : line) {
            if (isdigit(ch)) A.push_back(ch - '0');
            else if (isupper(ch)) A.push_back(ch - 'A' + 10);
            else if (islower(ch)) A.push_back(ch - 'a' + 36);
        }
        s = 0;
        mx = -1;
        for (int x : A){
            mx = max(mx, x);
            s += x;
        }
        ans = -1;
        for (int i = max(2, mx + 1) ; i <= 62; ++i) {
            if (s % (i - 1) == 0) {
                ans = i;
                break;
            }
        }
        cout << (ans == -1 ? "such number is impossible!" : to_string(ans)) << endl;
    }
    return 0;
}